packages/playwright-ct-vue/index.d.ts

```diff
@@ -41,26 +60,181 @@ type JsonObject = { [Key in string]?: JsonValue };
 
 type Slot = string | string[];
 
-export interface MountOptions<Props = Record<string, unknown>> {
-  props?: Props,
+interface MountOptions<Props> {
+  props?: (RawProps & Props) | ({} extends Props ? null : never);
   slots?: Record<string, Slot> & { default?: Slot };
   on?: Record<string, Function>,
   hooksConfig?: JsonObject,
 }
 
-interface MountResult<Props = Record<string, unknown>> extends Locator {
+interface MountResult<Props> extends Locator {
   unmount(): Promise<void>;
-  rerender(options: { props: Props }): Promise<void>
+  rerender(options: {
+    props: (RawProps & Props) | ({} extends Props ? null : never);
+  }): Promise<void>;
 }
 
+type RawProps = VNodeProps & {
+  __v_isVNode?: never;
+  [Symbol.iterator]?: never;
+} & Record<string, any>;
+
+type PublicProps = VNodeProps & AllowedComponentProps & ComponentCustomProps;
+
+type ComponentMountingOptions<T> = T extends DefineComponent<
+  infer PropsOrPropOptions,
+  any,
+  infer D,
+  any,
+  any
+>
+  ? MountOptions<
+      Partial<ExtractDefaultPropTypes<PropsOrPropOptions>> &
+        Omit<
+          Readonly<ExtractPropTypes<PropsOrPropOptions>> & PublicProps,
+          keyof ExtractDefaultPropTypes<PropsOrPropOptions>
+        >
+    > &
+      Record<string, any>
+  : MountOptions<any>;
+
+/**
+ * The types for mount have been copied directly from the VTU mount
+ * https://github.com/vuejs/vue-test-utils-next/blob/master/src/mount.ts.
+ *
+ * There isn't a good way to make them generic enough that we can extend them.
+ *
+ * In addition, we modify the types slightly.
+ *
+ * Changed MountingOptions<...> to MountOptions<...>
+ * and VueWrapper<...> to MountResult<...>.
+ */
```

Comment:
@pavelfeldman 
Added a comment to keep it a little easier to maintain. I'm not sure if there's a better way.. Seems like Cypress has taken the same approach.